File: nivel-04/ex-05/valor_repetido.py
# Encontra valor repetido dentro de um vetor

vetor = [1, 5, 8, 10, 8, 11]
repetidos = 0

i = 0
while i < len(vetor):         # Pegando um elemento do vetor
	for c in vetor[i+1:]:	  # Pegando uma comparação de i para frente
		if vetor[i] == c:     # Verificando igualdade
			repetidos += 1    # Se igual, soma em repetidos
	i += 1                    

if (repetidos > 0):        
	print('Elementos repetidos: {}'.format(repetidos))
else:
	print('Não existem elementos repetidos')

		
#assert repetido == True , 'existe vetor repetido' 

File: nivel-04/ex-05/valorRepetido.c
// Encontra valores repetidos em C

#include <stdio.h>
#include <assert.h>
#include <stdbool.h>

int encontraRepetido(int vet[], int tam){
	bool retorno;
	for(int i = 0; i < tam; i++){
		for(int j = i+1; j < tam; j++){
			if (vet[i] == vet[j]){
				return true;
				break;
			}else{
				retorno = false;
			}			
		}
	}
	return retorno;
}
void main(){

	int vetor[5] = {4, 2, 6, 2, 8};
	int vetor2[6] = {1, 2, 3, 4, 5, 6};

	assert(encontraRepetido(vetor, 5) == true);
	assert(encontraRepetido(vetor2,6) == false);

}